plio/io/isis_serial_number.py

```diff
@@ -96,9 +97,16 @@ def generate_serial_number(label):
                 serial_entry = search_translation[serial_entry]
             elif '*' in search_translation.keys() and search_translation['*'] != '*':
                 serial_entry = search_translation['*']
+            try:
+                serial_entry = serial_entry.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3]
```

Comment:
Without converting the datetime.datetime to string, the line
```
return '/'.join(serial_number)
```
fails because datetime objects don't have a join method.  It seems that .strftime's microsecond implementation includes 6 total digits (padded by 0's if insufficient digits are found).  In order to get the tests passing, I had to either modify the test or chop off the right-most digits.

Is this overfitting the logic to the tests, or is this the expected outcome?

Reply:
This is concerning. I'll have to look at how ISIS converts this because converting this differently could result in a different serial number which would be very bad.

Reply:
So ISIS doesn't interpret it as a date or time, it just interprets it as a string. I'm not sure how to recover the original string from either the PVL interface or the datetime itself.

For sure, we should not just trim the last 3 values from the milliseconds because different missions could use different precision on their time values.